Sources/MapboxMaps/Foundation/Camera/CameraAnimator.swift

```diff
@@ -74,27 +123,70 @@ public class CameraAnimator: NSObject {
+        animation = animations
     }
 
     /// Add a completion block to the animator. 
     public func addCompletion(_ completion: @escaping AnimationCompletion) {
-        propertyAnimator.addCompletion({ [weak self] animatingPosition in
-            guard let self = self else { return }
-            self.delegate?.schedulePendingCompletion(forAnimator: self, completion: completion, animatingPosition: animatingPosition)
-        })
+        let wrappedCompletion = wrapCompletion(completion)
+        propertyAnimator.addCompletion(wrappedCompletion)
+    }
+
+    internal func wrapCompletion(_ completion: @escaping AnimationCompletion) -> (UIViewAnimatingPosition) -> Void {
+        return { [weak self] animationPosition in
+            guard let self = self, let delegate = self.delegate else { return }
+            self.transition = nil // Clear out the transition being animated by this animator,
+                                  // since the animation is complete if we are here.
+            delegate.schedulePendingCompletion(forAnimator: self, completion: completion, animatingPosition: animationPosition)
+        }
     }
 
     /// Continue the animation with a timing parameter (`UITimingCurveProvider`) and duration factor (`CGFloat`).
     public func continueAnimation(withTimingParameters parameters: UITimingCurveProvider?, durationFactor: Double) {
         propertyAnimator.continueAnimation(withTimingParameters: parameters, durationFactor: CGFloat(durationFactor))
     }
+
+    public func update() {
+
+        // Only call jumpTo if this animator is currently "active" and there are known changes to animate.
+        guard propertyAnimator.state == .active,
+              let transition = transition,
+              let delegate = delegate else {
+            return
+        }
+
+        var cameraOptions = CameraOptions()
+        let interpolatedCamera = cameraView.localCamera
+
+        if transition.center.toValue != nil {
+            cameraOptions.center = interpolatedCamera.center?.wrap() // Wraps to [-180, +180]
+        }
+
+        if transition.bearing.toValue != nil {
+            cameraOptions.bearing = interpolatedCamera.bearing
+        }
+
+        // Honor the constant anchor if provided as part of the transition
+        if transition.constantAnchor != nil {
+            cameraOptions.anchor = transition.constantAnchor
+        } else if transition.anchor.toValue != nil {
+            cameraOptions.anchor = interpolatedCamera.anchor
+        }
+
+        if transition.padding.toValue != nil {
+            cameraOptions.padding = interpolatedCamera.padding
+        }
+
+        if transition.zoom.toValue != nil {
+            cameraOptions.zoom = interpolatedCamera.zoom
+        }
+
+        if transition.pitch.toValue != nil {
+            cameraOptions.pitch = interpolatedCamera.pitch
+        }
+
+        delegate.jumpTo(camera: cameraOptions)
```

Comment:
Let's only call `jumpTo` if at least 1 property (other than anchor) is non-nil. This will help to avoid the issue we have now where we perpetually repaint.

cc: @knov FYI

Reply:
I think the tighter scope of the CameraView already solves the issue. However, I'm not opposed to this as further safety.